tests/edges.test.js

```diff
@@ -42,4 +42,109 @@ test('can match top edge', () => {
   });
 });
 
-// TODO: middle, bottom, left, center, right
+test('can match middle edge', () => {
+  const edges = new Edges();
+
+  edges.add(new Box({ x: 100, y: 100, width: 200, height: 200 }));
```

Comment:
I'm thinking it would be worth making `x` and `y` slightly different from each other, as well as the `width` and `height`. Just so that there's no change of the test passing by accident of the opposing axis.

Reply:
Same for the other tests too.